### DISCUSSION TITLE
NextJS examples "with react helmet"

### DISCUSSION THREAD

**Discussion by Ravvsky** [vercel/next.js#41686] [Help]
I have no idea how to correctly change that file to typescript
https://github.com/vercel/next.js/blob/canary/examples/with-react-helmet/pages/_document.js

On line 9 I'm getting an error `A spread argument must either have a tuple type or be passed to a rest parameter.ts(2556)`

is there anywhere that file but typed

**Comment by thiswallz** [comment]
should be quite streight forward ,

pages/_document.tsx

``` typescript
import Document, { DocumentContext, DocumentInitialProps } from 'next/document'

class MyDocument extends Document {
  static async getInitialProps(
    ctx: DocumentContext
  ): Promise<DocumentInitialProps> {
    const initialProps = await Document.getInitialProps(ctx)

    return initialProps
  }
}

export default MyDocument
```
https://nextjs.org/docs/advanced-features/custom-document
